blackbox/docs/upgrading.txt

```diff
@@ -0,0 +1,122 @@
+this has been done, your tables will be forward compatible with future releases.
+
+To get started, upgrade your cluster like normal. When CrateDB 1.1 is running,
+you may now see that your cluster status is yellow rather than green.
+
+.. TODO: screenshot
+
+Depending on your situation, you may then be told that you should either
+migrate or reindex some of your tables.
+
+.. TODO: screenshot
+
+.. TODO: screenshot
+
+If your cluster status is green, skip to the section `Create New Snapshots`_.
+Otherwise, continue to the section `Preparation`_.
+
+Preparation
+------------
+
+Before you continue, use the :ref:`copy_to` command to create a one-off backup
+of each table. You can do this on any node in your cluster.
+
+Reindex Tables
+--------------
+
+For tables that need reindexing, use :ref:`ref-show-create-table` to back up
+the table schema in addition to the :ref:`copy_to` data backup you already
+created.
+
+Once you have both the table schema and table data backups, you can use
+:ref:`drop-table` to drop the table.
+
+Now, recreate the table using the backed up table schema.
+
+Once created, re-import your data using :ref:`copy_from`.
+
+When all tables that needed reindexing have been backed up, dropped, and
+recreated, you can proceed to the next section.
+
+Migrate Tables
+--------------
+
+To migrate your tables, you must shut down your cluster.
+
+Once your cluster is shut down, you must run the CrateDB migration tool on each
+of the tables you need to migrate. The CrateDB migration tool is bundled with
+the CrateDB distribution, and can be found at ``bin/migrate``.
+
+Test that the tool is working and is able to automatically pick up your local
+configuration by running this::
+
+    sh$ bin/migrate --dry-run
+
+If that doesn't work, consult the output of ``bin/migrate -h``.
+
+When the dry run looks okay, you can run the actual migration like so::
+
+    sh$ bin/migrate
+
+Optionally, you can run ``bin/migrate --verbose`` instead if you want more
+information about what the migration tool is doing.
+
+You must run ``bin/migrate`` on every node in your cluster.
+
+.. TIP::
+
+   If you run ``bin/migrate`` simultaneously on all nodes it will speed up the
+   overall migration process.
+
+Once ``bin/migrate`` has been run on every node in your cluster, you can start
+your cluster back up.
+
+.. WARNING::
+
+   Do not start your cluster back up until all tables have been migrated.
+
+When your cluster starts back up, check your node logs (configured via
+:ref:`conf-node-attributes`) for any important messages.
+
+Verify Upgrade
+--------------
+
+Open your CrateDB `admin console`_ and verify that your cluster status is now
+green, indicating that the migration was successful. If it is not, you may have
+missed some tables. Please go back to the section `Preparation`_.
+
+Create New Snapshots
+--------------------
+
+Before upgrading to CrateDB 1.2, you should create a new snapshot of every
```

Comment:
not before whatever, but after migration, because old snapshots won't be supported by new crate releases.